### PR TITLE
Explain function Max_Rootdepth

### LINKED ISSUE #164: Refactoring `Max_Rootdepth.m`
The file [Max_Rootdepth.m](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/src/Max_Rootdepth.m) has a few problems:

- Documentation is missing
 -  what does this function calculate?
 -  what happens in the for loop?
- Three global variables are used. These should instead be passed to the function
- Some commented out code is in the file. If this code is unused it should be removed

Additionally, it would be good to also rename the variables/structures according to the [style guide](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf) where possible.

### PR DESCRIPTION
**Pull request by EntingTang**
### Description
Please add a description of the changes proposed in the pull request.

### Update executable file
Please re-generate exe file if matlab codes are changed. About how to create an exe file, see [exe readme](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/exe/README.md).

### Checklist
- [ ] Add a reference to related issues.
- [ ] @mentions of the person or team responsible for reviewing proposed changes.
- [ ] This pull request has a descriptive title.
- [ ] Code is written according to the [guidelines](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf).
- [ ] The checks by [MISS_HIT style checker and linter](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/CONTRIBUTING.md#follow-matlab-style-guidelines), below the pull request, are successful (green).
- [ ] Documentation is available.
- [ ] Model runs successfully.

closes #164

### CODE REVIEW

File: src/Max_Rootdepth.m [hunk 1]
```diff
@@ -3,22 +3,29 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% REFERENCES
```
Comment (Crystal-szj) [inline src/Max_Rootdepth.m:5]:
```suggestion

```

File: src/Max_Rootdepth.m [hunk 2]
```diff
@@ -3,22 +3,29 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% REFERENCES
-function [bbx] = Max_Rootdepth(bbx, NL, KT, TT)
+
```
Comment (Crystal-szj) [inline src/Max_Rootdepth.m:6]:
```suggestion
```

File: src/Max_Rootdepth.m [hunk 3]
```diff
@@ -3,22 +3,29 @@
...
+    
     %%% OUTPUTS
     %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-    %%%%%%%%%%%%%%%%%%%%%% Root lenth distribution %%%%%%%%%%%%%%%%%%%%%%%%%
+    %%%%%%%%%%%%%%%%%%%%%% Root existence checking %%%%%%%%%%%%%%%%%%%%%%%%%
     %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-    R_depth = 350; %
-    Elmn_Lnth = 0;
+    R_depth = 350; % User-defined for different PFTs: maximum rooting depth [cm]
+    Elmn_Lnth = 0; %Rooting depth of each layer
```
Comment (Yunfei-Wang1993) [inline src/Max_Rootdepth.m:23]: comment body must be separated with whitespace from the starting %

File: src/Max_Rootdepth.m [hunk 4]
```diff
@@ -3,22 +3,29 @@
...
     for ML = 1:NL
         Elmn_Lnth = Elmn_Lnth + DeltZ(ML);
-        if Elmn_Lnth < Tot_Depth - R_depth
-            bbx(ML) = 0;
-        elseif Elmn_Lnth >= Tot_Depth - R_depth && Elmn_Lnth <= Tot_Depth - 5 % && TT(ML)>0
+        if Elmn_Lnth < Tot_Depth - R_depth %
```
Comment (Crystal-szj) [inline src/Max_Rootdepth.m:26]:
```suggestion
        if Elmn_Lnth < Tot_Depth - R_depth

```

**Review by Crystal-szj — CHANGES_REQUESTED** [review]
@EntingTang Thanks for your effort. Please see my comments in the code.

**Comment by SarahAlidoost** [comment]
@EntingTang @Crystal-szj Hi, are you still working on this pull request? can you please fix the conflict and finalize the pull request? so it can be merged after that. If you need help with fixing the "merge conflict", let me know.

**Review by Crystal-szj — APPROVED** [review]

**Comment by Crystal-szj** [comment]
Hi @SarahAlidoost, I've fixed the conflict and compared the results using [compare_git_branch_results.ipynb](https://github.com/Crystal-szj/STEMMUS_SCOPE/blob/main/test/compare_git_branch_results.ipynb). The results are same.

I would like to push the comparison results, but I am encountering [an issue with the passphrase](https://github.com/orgs/EcoExtreML/discussions/235) on my local computer, so I haven't been able to push the results yet.

**Comment by SarahAlidoost** [comment]
> Hi @SarahAlidoost, I've fixed the conflict and compared the results using [compare_git_branch_results.ipynb](https://github.com/Crystal-szj/STEMMUS_SCOPE/blob/main/test/compare_git_branch_results.ipynb). The results are same. Thus, the changes for the `Max_Rootdepth` function are ready to merge into the `main` branch.
> 
> I'm wondering if it is beneficial to push the comparison results as well. Currently, I am encountering [an issue with the passphrase](https://github.com/orgs/EcoExtreML/discussions/235) on my local computer, so I haven't been able to push the results yet.

Thanks for fixing the pull request. no need to run the test notebook or re-generate the exe file because the changes are about adding comments only. Feel free to merge.

**Pull request merged**